transport/src/main/java/io/netty/channel/DefaultChannelPipeline.java

```diff
@@ -1118,6 +1142,20 @@ private void callHandlerCallbackLater(AbstractChannelHandlerContext ctx, boolean
         }
     }
 
+    private void wakeUpForAutoFlush() {
+        if (autoFlush && ENQUEUE_WAKEUP_TASK_UPDATER.compareAndSet(this, 0, 1)) {
+            if (null == wakeupTask) {
+                wakeupTask = new Runnable() {
+                    @Override
+                    public void run() {
+                        ENQUEUE_WAKEUP_TASK_UPDATER.set(DefaultChannelPipeline.this, 0);
+                    }
+                };
+            }
+            channel().eventLoop().execute(wakeupTask);
+        }
+    }
+
```

Comment:
My minimal test case is a lot less minimal than I'd like, but it lives at https://gist.github.com/jchambers/75c665d566516290931a22b478e527f3. If you comment out [the line where we turn on `AUTO_FLUSH`](https://gist.github.com/jchambers/75c665d566516290931a22b478e527f3#file-http2client-java-L41), everything works great. If you leave it on, you wind up with an exception volcano. Will continue to dig.